### PR TITLE
feat: add debian variant

### PR DESCRIPTION
**Pull request by frezbo**
This adds a debian variant to be used as a base when building glibc based toolchain.

### CODE REVIEW

File: internal/pkg/convert/graph.go [hunk 1]
```diff
@@ -101,19 +101,46 @@ func (graph *GraphLLB) buildBaseImages() {
...
+			append(graph.commonRunOptions,
+				llb.Shlex("apk --no-cache --update add coreutils"),
+				llb.WithCustomName(graph.Options.CommonPrefix+"cksum-apkinstall"),
+			)...,
+		).Root()
+	case v1alpha2.Debian:
+		checksummer = llb.Image(
+			constants.DefaultDebianImage,
+			llb.WithCustomName(graph.Options.CommonPrefix+"cksum"),
+		).With()
```
Comment (frezbo) [inline internal/pkg/convert/graph.go:140]: `.With` is used since that only retuns a `llb.State` from `llb.Image` without no-op

Comment (smira) [reply internal/pkg/convert/graph.go:140]: this is a separate step, so I believe it doesn't have to be changed, it's a like a separate tree root which is independent of the variant.

**Comment by frezbo** [comment]
> I couldn't figure out how to set `HOST` and `BUILD` based on variant, since these are set in a `init` method, https://github.com/siderolabs/bldr/blob/main/internal/pkg/environment/platform.go#L65

@smira I'm still stuck here though

**Comment by smira** [comment]
> > I couldn't figure out how to set `HOST` and `BUILD` based on variant, since these are set in a `init` method, https://github.com/siderolabs/bldr/blob/main/internal/pkg/environment/platform.go#L65
> 
> @smira I'm still stuck here though

I don't think I have any "easy" answer for you here. I feel it needs to be refactored so that it becomes dependent on the variant?

**Comment by frezbo** [comment]
> > > I couldn't figure out how to set `HOST` and `BUILD` based on variant, since these are set in a `init` method, https://github.com/siderolabs/bldr/blob/main/internal/pkg/environment/platform.go#L65
> > 
> > 
> > @smira I'm still stuck here though
> 
> I don't think I have any "easy" answer for you here. I feel it needs to be refactored so that it becomes dependent on the variant?

I was not so sure about that, it seemed some parts implemented stuff from buildkit, it was confusing

**Comment by frezbo** [comment]
Fixed by using wolfi as toolchain https://github.com/siderolabs/extensions/pull/184

**Pull request closed without merging**